File: algorithms/math/recursion/factorial.js
// Problem: Give an integer 'n', find the factorial (n!) of that integer
// factorial(4) = 4*3*2*1 = 24

function recursiveFactorial(n) {
  // base case
  if (n === 0) {
    return 1;
  }
  return n * recursiveFactorial(n - 1);
}

console.log(recursiveFactorial(0)); // 1
console.log(recursiveFactorial(1)); // 1
console.log(recursiveFactorial(5)); // 120

// Big-O = O(n)

File: algorithms/math/python/recursion/factorial.py
# O(n^2)
def recursive_factorial(n):
    # base case
    if n == 0:
        return 1
    return n * recursive_factorial(n - 1)


print(recursive_factorial(0))  # 1
print(recursive_factorial(1))  # 1
print(recursive_factorial(5))  # 120
